docs/samples/gcp-iap/README.md

```diff
@@ -0,0 +1,110 @@
+2. Your gcloud config is initialised to the project containing the k8s cluster and has a service-account that can download IAP key file.
+3. You are using Knative serving v0.11.2 or v0.14.0+
+4. You are using a recent version of KFServing (v0.3+)
+
+To ensure your cluster has the **knative istio probe fix**, you can use [kfctl_gcp_iap.v1.0.2.yaml](https://raw.githubusercontent.com/kubeflow/manifests/v1.0-branch/kfdef/kfctl_gcp_iap.v1.0.2.yaml), editing the repos section in the yaml before deploying to your Kubernetes cluster.
+```
+    repos:
+    - name: manifests
++     uri: https://github.com/kubeflow/manifests/archive/master.tar.gz
+-     uri: https://github.com/kubeflow/manifests/archive/v1.0.2.tar.gz
+-   version: v1.0.2
+```
+
+Once IAP is enabled and configured, your k8s cluster should have a GCP external http loadbalancer protected by IAP.
+
+### Overview
+This example shows how the [sklearn iris sample](https://github.com/kubeflow/kfserving/tree/master/docs/samples/sklearn) can be adapted to support GCP IAP.  We will use a pre-trained model that is hosted on a public gcs bucket.  The model predicts a binary classificaton of iris specie using 4 numerical features.
+
+In this example we will:
+ 1. **Enable GCP IAP**
+ 1. **Change firewall rules (for private k8s clusters)**
+ 1. **Create the inference service**
+ 1. **Test the private predict endpoint (using port-forwarding)**
+ 1. **Expose the inference service externally using an additional Istio Virtual Service**
+ 1. **Test the external predict endpoint (using iap_request.py)**
+
+Step 5. will become unnecessary when this [kfserving issue](https://github.com/kubeflow/kfserving/issues/824) is resolved.
+
+### Enable GCP IAP for k8s
+You have followed the steps to [Set up OAuth for Cloud IAP](https://www.kubeflow.org/docs/gke/deploy/oauth-setup/)
+
+We assume that you have created Kubeflow profiles and that [GCP workload identity](https://www.kubeflow.org/docs/gke/authentication/) is configured.
+
+This may not be essential for this example, but is recommended as it maps GCP service-accounts (understood by the IAP) to kubernetes service-accounts (understood by kubernetes and istio RBAC).
+
+### Change firewall rules (for private k8s clusters) 
+If you are running Kubeflow in a [private GKE cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters) it will be configured with a restrictive firewall.  
+You will need to ensure that the k8s master node(s) can talk to the following k8s worker node(s) ports: 
+ - 6443 (for cert-manager) 
+ - 8443 (for istio readiness probe)
+
+To create these rules firewall rules, modify and run:
+
+```
+gcloud compute firewall-rules create kubeflow-webhook-probe --allow=tcp:8443 --target-tags kubeflow-worker --direction INGRESS --network default --priority 1000 --source-ranges 172.16.0.0/28
```

Comment:
tcp:6443?

Reply:
Good spot I've changed line 51 (cert-manager) to open port 6443